### PR TITLE
chore: enable request logging when SnykCode logger is in debug mode [ROAD-843]

### PR DESCRIPTION
**Pull request by bastiandoetsch**
### Description

This PR enables HTTP request logging for the Snyk Code Client when the `SnykCode` logger is set to debug and enables to monitor / check API communication, thus.

### Checklist

- [ x ] Tests all succeed
- [ x ] Linted
- [ x ] CHANGELOG.md updated
- [ x ] README.md updated, if user-facing

### CODE REVIEW

**Review by ArtsiomCh — CHANGES_REQUESTED** [review]
Would be happy and not create the ticket if fix could be such an easy thing to do... :)

- We re-create `retrofit` from few other places, so that `requestLogin` param will be overridden to the default `false` there. See `ai.deepcode.javaclient.core.DeepCodeParamsBase#setApiUrl()` usages.
- Not sure where `HttpLoggingInterceptor` will output. Most likely to STDOUT and that probably(?) will go to Idea log as well, but need to be checked. 
- Don't think we want to see such a detailed logs (at least I see in failed tests it's very verbose) every time we have DEBUG level enabled for the snyk classes. Would prefer to have separate feature flag in Idea's Registry for that.
- also probably some tests need to be added...

**Comment by bastiandoetsch** [comment]
moving to slack...

File: .snyk [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Snyk (https://snyk.io) policy file, patches or ignores known vulnerabilities.
+version: v1.25.0
+# ignores vulnerabilities until expiry date; change duration by modifying expiry date
+ignore:
+  '*':
+    - build/*:
+        reason: None Given
+        expires: 2032-05-20T07:12:38.106Z
+        created: 2022-04-20T07:12:38.108Z
+    - src/integTest/*:
```
Comment (ArtsiomCh) [inline .snyk:10]: do we really want to ignore any vulns in our tests? may be just in resources...

File: .snyk [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        reason: None Given
+        expires: 2022-05-20T07:13:02.819Z
+        created: 2022-04-20T07:13:02.822Z
+    - src/test/*:
```
Comment (ArtsiomCh) [inline .snyk:14]: see above

File: src/main/kotlin/io/snyk/plugin/snykcode/core/SnykCodeParams.kt [hunk 3]
```diff
@@ -19,10 +20,30 @@ class SnykCodeParams private constructor() : DeepCodeParamsBase(
     { getWaitForResultsTimeout() },
     codeRestApi
 ) {
+    private val requestLogging = Logger.getInstance(SCLogger.presentableName + "RequestLogging").isDebugEnabled
```
Comment (ArtsiomCh) [inline src/main/kotlin/io/snyk/plugin/snykcode/core/SnykCodeParams.kt:23]: that should be `fun()` (or val with `get()`) cause we need to re-read setting every time it's called out (user can change it after class created).

**Review by ArtsiomCh — CHANGES_REQUESTED** [review]
Some (easy) changes needed
Also please rebase CHANGELOG to resolve merge conflict

**Review by ArtsiomCh — APPROVED** [review]
LGTM
Good job!

**Pull request merged**